docs/modules/tls.md

```diff
@@ -321,6 +321,55 @@ The alternative approach is easier for development, and that is to supply the PE
 will store the certificate into the flash chip and turn on verification for that certificate. Subsequent boots of the nodemcu can then
 use `tls.cert.verify(true)` and use the stored certificate.
 
+## tls.cert.auth()
+
+Controls the certificate verification process when the Nodemcu authenticates against the client like when receiving a secure connection.
+
+#### Syntax
+`tls.cert.auth(enable)`
+
+`tls.cert.auth(pemdata[, pemdata])`
+
+#### Parameters
+- `enable` A boolean which indicates whether verification should be enabled or not. The default at boot is `false`.
+- `pemdata` A string containing the CA certificate to use for verification. There can be several of these.
+
+#### Returns
+`true` if it worked.
+
+Can throw a number of errors if invalid data is supplied.
+
+#### Example
+Open an mqtt client.
+```
+tls.cert.auth(true)
+tls.cert.verify(true)
+
+m = mqtt.Client('basicPubSub', 1500, "admin", "admin", 1)
+```
+For further discussion see https://github.com/nodemcu/nodemcu-firmware/issues/2576
+
+Load a certificate into the flash chip and make a request.
+
+```
+tls.cert.auth([[
+-----BEGIN CERTIFICATE-----
+CLIENT CERTIFICATE String (PEM file)
+-----END CERTIFICATE-----
+]])
+```
+
+#### Notes
+The certificate needed for verification is stored in the flash chip. The `tls.cert.auth` call with `true`
+enables verification against the value stored in the flash.
+
```

Comment:
Calling this verification is probably confusing; the server is doing the verifying, the client (the NodeMCU device) is doing the proving.

Reply:
Thanks for this clarification.. As I mentioned I don't really know much about this stuff. Tried to change the sentence, but probably still failed.
@nwf  you are also welcome to directly edit this PR if you like.

Reply:
How about "The client certificate the server requests to verify the callers authenticity is stored in the flash chip."

> you are also welcome to directly edit this PR if you like

Looks like you didn't set the respective flag when you created the PR. I can't edit it.